cmd/oras/root/manifest/fetch.go

```diff
@@ -72,9 +74,22 @@ Example - Fetch raw manifest from an OCI layout archive file 'layout.tar':
 `,
 		Args: oerrors.CheckArgs(argument.Exactly(1), "the manifest to fetch"),
 		PreRunE: func(cmd *cobra.Command, args []string) error {
-			if opts.outputPath == "-" && opts.OutputDescriptor {
-				return errors.New("`--output -` cannot be used with `--descriptor` at the same time")
```

Comment:
Any table drive code like table tests I find a lot harder to debug because if there is a problem, you aren't sure which item you were processing when the problem occurred. It also tends to be more lines of code then just performing each action.

Reply:
How about 
```golang
switch {
case opts.outputPath == "-" && opts.Template != "":
  return fmt.Errorf("`--output -` cannot be used with `--format` at the same time")
case opts.OutputDescriptor && opts.Template != "" :
  return fmt.Errorf("`--descriptor` cannot be used with `--format` at the same time")
case opts.OutputDescriptor && opts.outputPath != "":
  return fmt.Errorf("`--descriptor` cannot be used with `--output` at the same time")
}
```

BTW I will need change to tabulated checking when the number of exclusive flags increase to 4 anyway

Reply:
`--descriptor` can be used with `--output` but cannot be used with `--output -`.

Reply:
Is there any reason that `--descriptor` cannot be used with `--format`?

Reply:
> Is there any reason that `--descriptor` cannot be used with `--format`?

Both takes stdout and will corrupt each other.